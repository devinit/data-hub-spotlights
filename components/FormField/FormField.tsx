import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

const FormField: FunctionComponent<{ className?: string }> = ({ className, children }) => (
  <div className={classNames('form-field', className)}>{children}</div>
);

FormField.defaultProps = { className: 'form-field--inline-block' };

export { FormField };
