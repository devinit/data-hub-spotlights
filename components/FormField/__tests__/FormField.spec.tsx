/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { FormField } from '../FormField';

describe('FormField', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<FormField />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children correctly', () => {
    const renderer = TestRenderer.create(
      <FormField>
        <div>My Child</div>
      </FormField>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
