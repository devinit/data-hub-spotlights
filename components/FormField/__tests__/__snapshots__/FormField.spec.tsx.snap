// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders correctly 1`] = `
<div
  className="form-field form-field--inline-block"
/>
`;

exports[`FormField renders with children correctly 1`] = `
<div
  className="form-field form-field--inline-block"
>
  <div>
    My Child
  </div>
</div>
`;
